uefi/src/proto/debug/mod.rs

```diff
@@ -170,3 +170,46 @@ pub enum ProcessorArch: u32 => {
     /// RISC-V 128-bit
     RISCV_128   = 0x5128,
 }}
+
+/// The debug port protocol abstracts the underlying debug port
+/// hardware, whether it is a regular Serial port or something else.
+#[repr(C)]
+#[unsafe_protocol("eba4e8d2-3858-41ec-a281-2647ba9660d0")]
+pub struct DebugPort {
+    reset: extern "efiapi" fn(this: &DebugPort) -> Status,
+    write: extern "efiapi" fn(
+        this: &DebugPort,
+        timeout: u32,
+        buffer_size: &mut usize,
+        buffer: *const c_void,
+    ) -> Status,
+    read: extern "efiapi" fn(
+        this: &DebugPort,
+        timeout: u32,
+        buffer_size: &mut usize,
+        buffer: *mut c_void,
+    ) -> Status,
+    poll: extern "efiapi" fn(this: &DebugPort) -> Status,
+}
+
+impl DebugPort {
+    /// Flushes the port.
```

Comment:
Change doc to "Resets the debugport device." With "flush" I usually think of writing out remaining buffered bytes.